screenapiexample/src/main/java/net/gini/android/vision/screen/CameraScreenApiActivity.java

```diff
@@ -1,27 +1,159 @@
+        InputStream inputStream = null;
+        try {
+            inputStream = getContentResolver().openInputStream(uri);
+            if (inputStream != null) {
+                final byte[] magicBytes = new byte[4];
+                final int read = inputStream.read(magicBytes);
+                inputStream.reset();
+                return read != -1
+                        && (isJpegWithExif(inputStream, magicBytes, read)
+                        || isPDF(uri, magicBytes, read));
+            }
+        } catch (FileNotFoundException e) {
+            LOG.error("Could not open document", e);
+            return false;
+        } catch (IOException e) {
+            LOG.error("Could not read document", e);
+            return false;
+        } finally {
+            if (inputStream != null) {
+                try {
+                    inputStream.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean isJpegWithExif(final InputStream inputStream, final byte[] magicBytes,
+            final int read)
+            throws IOException {
+        return isDecodableToBitmap(inputStream)
+                && read == 4
+                && magicBytes[0] == (byte) 0xFF
+                && magicBytes[1] == (byte) 0xD8
+                && magicBytes[2] == (byte) 0xFF
+                && magicBytes[3] == (byte) 0xE1;
+    }
+
+    private boolean isDecodableToBitmap(final InputStream inputStream)
+            throws IOException {
+        BitmapFactory.Options options = new BitmapFactory.Options();
+        final Bitmap bitmap = BitmapFactory.decodeStream(inputStream, new Rect(), options);
+        return bitmap != null;
+    }
+
+    private boolean isPDF(final Uri uri, final byte[] magicBytes, final int read) {
+        boolean isRenderablePdf = true;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            isRenderablePdf = isRenderablePDF(uri);
+        }
+        return isRenderablePdf
+                && read == 4
+                && magicBytes[0] == (byte) 0x25
+                && magicBytes[1] == (byte) 0x50
+                && magicBytes[2] == (byte) 0x44
+                && magicBytes[3] == (byte) 0x46;
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    private boolean isRenderablePDF(final Uri uri) {
+        final ParcelFileDescriptor fileDescriptor;
+        try {
+            fileDescriptor = getContentResolver().openFileDescriptor(uri, "r");
+        } catch (FileNotFoundException e) {
+            LOG.error("Pdf not found", e);
+            return false;
+        }
+        if (fileDescriptor == null) {
+            LOG.error("Pdf not found");
+            return false;
+        }
+        try {
+            final PdfRenderer pdfRenderer = new PdfRenderer(fileDescriptor);
+            pdfRenderer.close();
+            return true;
+        } catch (IOException e) {
```

Comment:
Why don't you combine both try-catch?

Reply:
To keep the exception catching close to the related throwing method. Otherwise it wouldn't be obvious which method throws which exception...